src/coreclr/tools/superpmi/superpmi/neardiffer.cpp

```diff
@@ -318,147 +341,91 @@ struct DiffData
-    if (data->cr2->CallTargetTypes->GetIndex((DWORDLONG)Offset1b) != -1)
-    {
-        // This logging is too noisy, so disable it.
-        // LogVerbose("Found VSD callsite, did softer compare than ideal");
-        return true;
-    }
-    if (data->cr2->CallTargetTypes->GetIndex((DWORDLONG)Offset2b) != -1)
-    {
-        // This logging is too noisy, so disable it.
-        // LogVerbose("Found VSD callsite, did softer compare than ideal");
-        return true;
-    }
+        if ((iaddr1 < iaddr1end) &&
+            GetArm32MovwConstant(iaddr1, &reg1_w, &con1_w))
+        {
+            if ((iaddr1 + 1 < iaddr1end) &&
+                GetArm32MovtConstant(iaddr1 + 1, &reg1_t, &con1_t) &&
+                (reg1_w == reg1_t))
+            {
+                addr1 = con1_w + (con1_t << 16);
+                pair1 = true;
+            }
+        }
 
-    // Case might be a field address that we handed out to handle inlined values being loaded into
-    // a register as an immediate value (and where the address is encoded as an indirect immediate load)
-    size_t realTargetAddr = (size_t)data->cr2->searchAddressMap((void*)gOffset2);
-    if (realTargetAddr == gOffset1)
-        return true;
+        // Look for a movw/movt address pattern in code stream 2.
 
-    // Case might be a field address that we handed out to handle inlined values being loaded into
-    // a register as an immediate value (and where the address is encoded and loaded by immediate into a register)
-    realTargetAddr = (size_t)data->cr2->searchAddressMap((void*)offset2);
-    if (realTargetAddr == offset1)
-        return true;
-    if (realTargetAddr == 0x424242) // this offset matches what we got back from a getTailCallCopyArgsThunk
-        return true;
+        if ((iaddr2 < iaddr2end) &&
+            GetArm32MovwConstant(iaddr2, &reg2_w, &con2_w))
+        {
+            if ((iaddr2 + 1 < iaddr2end) &&
+                GetArm32MovtConstant(iaddr2 + 1, &reg2_t, &con2_t) &&
+                (reg2_w == reg2_t))
+            {
+                addr2 = con2_w + (con2_t << 16);
+                pair2 = true;
+            }
+        }
 
-    realTargetAddr = (size_t)data->cr2->searchAddressMap((void*)(gOffset2));
-    if (realTargetAddr != (size_t)-1) // we know this was passed out as a bbloc
-        return true;
+        if (pair1 && pair2 && (reg1_w == reg2_w))
+        {
+            // Use the constructed constant instead.
+            *offset1 = addr1;
+            *offset2 = addr2;
+            *skip1 = *skip2 = 1; // skip the `movt` instruction
+            return true;
+        }
+    }
 
-    // A new clause that tries to handle the case where neither cr1 or cr2 is the
-    // recorded CR (diff case with two jits, neither of which is the one used for
-    // collection)
+#if 0
+    // We might want to do something similar for mov/movk/movk/movk sequences on Arm64. The following code was
+    // previously used, but currently is not active.
```

Comment:
Delete the code?

Reply:
I'm not convinced yet that we don't need (or will never need) this, so I think it's worth leaving.